FogCreek/Glitch-Community
# Garrulous literature: show top results on search page

## Links
* Remix link: https://garrulous-literature.glitch.me
* Manuscript link: https://glitch.manuscript.com/f/cases/3328158/
* Specs/Notion docs: https://www.notion.so/glitch/Search-244881f18e794507b67ef082a101cd84#0e64be9904114e8b9fd974e7def4ae4f (scroll to "Top Results")

## GIF/Screenshots
![Screen Shot 2019-04-04 at 1 05 59 PM](https://user-images.githubusercontent.com/938722/55574456-71959b00-56da-11e9-818a-af2880355ac6.png)

## Changes
* add a "Top Results" section to the search results page
* show exact matches from search results in top results
* componentize & storybook SearchResults, Loading, NotFound

## How To Test
* search for users, projects, eg. https://garrulous-literature.glitch.me/search?q=react
* exact matches (for name, login, etc) should be in top results, in addition to the regular list
* if the top result is an exact match, and there are no further results in that type, then don't show the type section in All Results (see Edge Cases in Notion spec)

## Feedback I'm looking for
* does this render correctly?
* are there any edge cases I am missing?
